extensions/wikia/PortableInfoboxBuilder/tests/PortableInfoboxBuilderServiceTest.php

```diff
@@ -51,6 +58,15 @@ public function testUpdateDocumentation($data, $expected) {
 			$data['newDoc'], $data['oldContent']));
 	}
 
+
+	public function infoboxArrayValidityDataProvider() {
+		return [
+			[ [], true, 'Empty infobox array is not valid' ],
+			[ [ '<infobox><data source="source1"></data></infobox>' ], true, 'Single infobox array is valid' ],
+			[ [ '<infobox><data source="source1"></data></infobox>', '<infobox><data source="source2"></data></infobox>' ], false, 'Multi- infobox array is not valid' ],
+		];
```

Comment:
`isValidInfoboxArray` does this check:
```
return empty( $infoboxes ) || ( count( $infoboxes ) === 1 && $this->isSupportedMarkup( $infoboxes[0] ) );
```

If this is supposed to be a proper unit test then we should test against these conditions. Also, `isSupportedMarkup` result should be mocked and passed from provider. Then you wouldn't say we have separate tests for that ;)